lib/_http_common.js

```diff
@@ -290,11 +289,12 @@ exports._checkIsHttpToken = checkIsHttpToken;
  *  field-vchar    = VCHAR / obs-text
  **/
 function checkInvalidHeaderChar(val) {
-  val = '' + val;
+  if (typeof val !== 'string')
+    val = '' + val;
   for (var i = 0; i < val.length; i++) {
     const ch = val.charCodeAt(i);
-    if (ch === 9) continue;
-    if (ch <= 31 || ch > 255 || ch === 127) return true;
+    if (ch !== 9 && (ch <= 31 || ch > 255 || ch === 127))
```

Comment:
Why is this change from `continue` to `return` OK? It looks like it changes the meaning of the above code - if it does so intentionally would you mind adding a test case?